packages/atomic/src/components/atomic-result-template/atomic-result-template.tsx

```diff
@@ -6,6 +6,22 @@ import {
 } from '@coveo/headless';
 import {MapProp} from '../../utils/props-utils';
 
+const resultSectionTags = [
+  'atomic-result-section-visual',
+  'atomic-result-section-badges',
+  'atomic-result-section-actions',
+  'atomic-result-section-title',
+  'atomic-result-section-title-metadata',
+  'atomic-result-section-emphasized',
+  'atomic-result-section-excerpt',
+  'atomic-result-section-bottom-metadata',
+];
+
+export interface TemplateContent {
+  innerHTML: string;
+  usesSections: boolean;
+}
```

Comment:
A result list can contain a mix of result templates with and without sections. I could've made `usesSections` a property of the result template, but detecting whether result sections are part of the template is a pretty solid approach too and reduces configuration.

Reply:
I think auto-detecting and reducing config makes a lot of sense 👍